From 9cc20d34284cf46ef8af0fe1d43802c79e3560e1 Mon Sep 17 00:00:00 2001
From: Liu Changcheng <changcheng.liu@intel.com>
Date: Wed, 11 Feb 2015 10:58:01 +0800
Subject: [PATCH 271/273] atomisp:avoid memory leak in atomisp_set_parameters

memory leak bug when free css_param

Change-Id: I8e75b6f8eb615bbd0aaa755f101f4edc2f1362f8
Tracked-On: https://jira01.devtools.intel.com/browse/IMINAN-21568
Signed-off-by: Jun Zhang <jun.zhang@intel.com>
Signed-off-by: Changcheng Liu <changcheng.liu@intel.com>
Signed-off-by: Ding Yunliang <yunliang.ding@intel.com>
Reviewed-on: https://icggerrit.ir.intel.com/41361
Reviewed-by: Jing Tao <jing.tao@intel.com>
Build: Wei Tao <wei.tao@intel.com>
Approver: Bingbu Cao <bingbu.cao@intel.com>
Reviewed-by: Bingbu Cao <bingbu.cao@intel.com>
Integrator: Wei Tao <wei.tao@intel.com>
Approver: Wei Tao <wei.tao@intel.com>
Reviewed-by: Wei Tao <wei.tao@intel.com>
Maintainer: Wei Tao <wei.tao@intel.com>
---
 .../camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_cmd.c   | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_cmd.c b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_cmd.c
index 3f0e9d6..bc1bd2e 100644
--- a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_cmd.c
+++ b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_cmd.c
@@ -3605,7 +3605,10 @@ int atomisp_set_parameters(struct video_device *vdev,
 
 apply_parameter_failed:
 	if (css_param)
-		atomisp_kernel_free(css_param);
+		atomisp_free_css_parameters(css_param);
+	if (param)
+		atomisp_kernel_free(param);
+
 	return ret;
 }
 
-- 
1.9.1

